lib/grape/middleware/logger.rb

```diff
@@ -91,6 +93,18 @@ def parameters
     end
   end
 
+  def headers
+    request_headers = env[Grape::Env::GRAPE_REQUEST_HEADERS].to_hash
+    return request_headers if @options[:headers] == :all
+
+    headers_needed = Array(@options[:headers])
+    Hash[request_headers.sort].select do |key, value|
```

Comment:
Instead of iterating over all headers, what about just iterating over the `headers_needed` (which will likely be less)?